File: Discount/userpromo.go
package main

import "time"

type UserPromo struct {
	ID            string    `bson:"id"`
	CustomerId    string    `bson:"customerId"`
	CustomerEmail string    `bson:"customerEmail"`
	Promo         float64   `bson:"promo"`
	ValidUntil    time.Time `bson:"validUntil"`
}

File: WebUI/Models/Discounts/UserPromo.cs
namespace WebUI.Models.Discounts
{
    public record UserPromo
    {
        public Guid Id { get; set; }

        public Guid CustomerId { get; set; }

        public required string CustomerEmail { get; set; }

        public decimal Cashback { get; set; }

        public DateTime ValidUntil { get; set; }
    }
}
